src/object/srv_ec_aggregate.c

```diff
@@ -1628,30 +1630,25 @@ agg_process_holes_ult(void *arg)
 	}
 
 	agg_param = container_of(entry, struct ec_agg_param, ap_agg_entry);
+	rc = pool_map_find_failed_tgts(agg_param->ap_pool_info.api_pool->sp_map,
+				       &targets, &failed_tgts_cnt);
+	if (rc) {
+		D_ERROR(DF_UOID" pool_map_find_failed_tgts failed: "
+			DF_RC"\n", DP_UOID(entry->ae_oid), DP_RC(rc));
+		goto out;
+	}
+
 	/* Invoke peer re-replicate */
 	for (peer = 0; peer < p; peer++) {
 		if (pidx == peer)
 			continue;
-		rc = pool_map_find_failed_tgts(
-			agg_param->ap_pool_info.api_pool->sp_map,
-			&targets, &failed_tgts_cnt);
-		if (rc) {
-			D_ERROR(DF_UOID" pool_map_find_failed_tgts failed: "
-				DF_RC"\n", DP_UOID(entry->ae_oid), DP_RC(rc));
-			goto out;
-		}
 
-		if (targets != NULL) {
-			for (i = 0; i < failed_tgts_cnt; i++) {
-				if (targets[i].ta_comp.co_rank ==
-				    entry->ae_peer_pshards[peer].sd_rank) {
-					D_ERROR(DF_UOID" peer %d parity tgt "
-						"failed\n",
-						DP_UOID(entry->ae_oid),
-						peer);
-					rc = -1;
-					goto out;
-				}
+		for (i = 0; targets && i < failed_tgts_cnt; i++) {
+			if (targets[i].ta_comp.co_rank == entry->ae_peer_pshards[peer].sd_rank) {
+				D_ERROR(DF_UOID" peer %d parity tgt failed\n",
+					DP_UOID(entry->ae_oid), peer);
```

Comment:
Not this patch issue, but this probably should not be error message(DB_EPC maybe), otherwise it will output a lot error message during degrade mode.